include/maths/CBasicStatistics.h

```diff
@@ -1327,10 +1330,18 @@ class MATHS_EXPORT CBasicStatistics {
 
     public:
         explicit COrderStatisticsHeap(std::size_t n, const LESS& less = LESS{})
-            : TImpl{std::vector<T>(n, T{}), less} {}
+            : TImpl{std::vector<T>(std::max(n, std::size_t(1)), T{}), less} {
+            if (n == 0) {
+                LOG_ERROR(<< "Invalid size of 0 for statistics accumulator");
+            }
+        }
 
         //! Reset the number of statistics to gather to \p n.
         void resize(std::size_t n) {
+            if (n == 0) {
+                LOG_ERROR(<< "Invalid resize of 0 for statistics accumulator");
```

Comment:
nit: I think this reads slightly better as `Invalid resize to 0 for order statistics accumulator`.